app/gui/src/presenter/project.rs

```diff
@@ -464,7 +479,16 @@ impl Project {
         view: view::project::View,
         status_bar: view::status_bar::View,
     ) -> FallibleResult<Self> {
+        debug!("Initializing project controller...");
         let init_result = controller.initialize().await?;
-        Ok(Self::new(ide_controller, controller, init_result, view, status_bar))
+        debug!("Project controller initialized.");
+        let presenter =
+            Self::new(ide_controller, controller.clone(), init_result, view, status_bar);
+        debug!("Project presenter created.");
+        // Following the project initialization, the Undo/Redo stack should be empty.
+        // This makes sure that any initial modifications resulting from the GUI initialization
+        // won't clutter the undo stack.
+        controller.model.urm().repository.clear_all();
+        Ok(presenter)
```

Comment:
It looks a bit like a workaround. Cannot we rather create a transaction in this very function?

Reply:
During a call, we concluded this is the best way for now.